### PR TITLE
Implement TelnetFallback entry point to Netmiko

### PR DESCRIPTION
**Pull request by ktbyers**
Will try to connect using telnet if SSH connection fails for the given device type.

### CODE REVIEW

**Comment by ktbyers** [comment]
FYI @cla-azzarello

**Comment by ktbyers** [comment]
@cla-azzarello This should be very similar to what you implemented. I renamed it `TelnetFallback` (people almost never agree on names).

I also made it only: `try SSH -> then try telnet` (i.e. there is no try telnet and then SSH).

Basically, if you have SSH working, you should not be using telnet.

I also added some minor logic to handle device_types of the form `cisco_ios_ssh` and have them get converted to `cisco_ios_telnet`.

Thanks for all your work on this.

**Comment by ktbyers** [comment]
Let me know if you see any issues. @cla-azzarello 

I am going to go ahead and merge this, but if there are issues, then we can make changes/fixes.

**Comment by cla-azzarello** [comment]
Good point @ktbyers !

Thank you!

File: netmiko/__init__.py [hunk 1]
```diff
@@ -66,6 +67,7 @@ def check_python_version():  # type: ignore
 
 __all__ = (
     "ConnectHandler",
+    "AgnosticHandler",
```
Comment (cla-azzarello) [inline netmiko/__init__.py:70]: @ktbyers this should be TelnetFallback

Comment (ktbyers) [reply netmiko/__init__.py:70]: I will fix this.

**Pull request merged**